systems/init/lookup/cmd/lookup/main.go

```diff
@@ -71,23 +67,51 @@ func initConfig() {
 		}
 	}
 
-	log.Debugf("\nService: %s DB Config: %+v", internal.ServiceName, serviceConfig.DB)
+	log.Debugf("\nService: %s DB Config: %+v Service: %+v MsgClient Config %+v", internal.ServiceName, serviceConfig.DB, serviceConfig.Service, serviceConfig.MsgClient)
 
 	internal.IsDebugMode = serviceConfig.DebugMode
 }
 
 func runGrpcServer(d sql.Db) {
+
 	instanceId := os.Getenv("POD_NAME")
+	if instanceId == "" {
+		/* used on local machines */
+		inst, err := uuid.NewV4()
+		if err != nil {
+			log.Fatalf("Failed to genrate instanceId. Error %s", err.Error())
+		}
+		instanceId = inst.String()
+	}
 
-	//var mbClient *mb.MsgBusClient
 	mbClient := mb.NewMsgBusClient(serviceConfig.MsgClient.Timeout, internal.SystemName,
 		internal.ServiceName, instanceId, serviceConfig.Queue.Uri,
-		serviceConfig.MsgClient.Host, serviceConfig.MsgClient.RetryCount,
-		serviceConfig.MsgClient.ListnerRoutes)
+		serviceConfig.Service.Uri, serviceConfig.MsgClient.Host, serviceConfig.MsgClient.Exchange,
+		serviceConfig.MsgClient.ListenQueue, serviceConfig.MsgClient.PublishQueue,
+		serviceConfig.MsgClient.RetryCount,
+		serviceConfig.MsgClient.ListenerRoutes)
+
+	log.Debugf("MessageBus Client is %+v", mbClient)
+
 	grpcServer := ugrpc.NewGrpcServer(*serviceConfig.Grpc, func(s *grpc.Server) {
 		srv := server.NewLookupServer(db.NewNodeRepo(d), db.NewOrgRepo(d), db.NewSystemRepo(d), mbClient)
+		nSrv := server.NewLookupEventServer(db.NewNodeRepo(d), db.NewOrgRepo(d), db.NewSystemRepo(d))
 		generated.RegisterLookupServiceServer(s, srv)
+		egenerated.RegisterEventNotificationServiceServer(s, nSrv)
 	})
 
+	go msgBusListener(mbClient)
+
 	grpcServer.StartServer()
 }
+
+func msgBusListener(m mb.MsgBusServiceClient) {
+
+	if err := m.Register(); err != nil {
+		log.Fatalf("Failed to register to Message Client Service. Error %s", err.Error())
```

Comment:
Use logrus instead

Reply:
log is uased as alias for logrus. For this and all other comments regarding log.